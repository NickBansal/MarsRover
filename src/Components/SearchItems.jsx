import React from 'react'
import '../Stylesheets/SearchPage.css'
import { Link } from '@reach/router'
import NotAvailable from '../Stylesheets/Images/not-available.png';

const addDefaultSrc = (event) => {
    event.target.src = `${NotAvailable}`
}

const SearchItems = ({ allItems, handleClick }) => {
    return (
        <div>
            {allItems.length > 0 &&
                <div className="Results">
                    <form onClick={(e) => handleClick(e.target.value)}>
                        <div className="RadioSelect">
                            <input type="radio" name="gender" value="image" /> Image
                        </div>
                        <div className="RadioSelect">
                            <input type="radio" name="gender" value="video" /> Video
                        </div>
                        <div className="RadioSelect">
                            <input type="radio" name="gender" value="audio" /> Audio
                        </div>
                    </form>
                    <p>Results: {allItems.length}</p>
                </div>}
            {allItems.map(items => {
                return (
                    <Link
                        key={items.data[0].nasa_id}
                        to={`/search/${items.data[0].nasa_id}`}>
                        <div
                            className="SingleThumbnail">
                            <img
                                onError={addDefaultSrc}
                                src={items.links[0].href}
                                alt={items.data[0].title} />
                            <strong><p>{items.data[0].title.substring(0, 40)}</p></strong>
                        </div>
                    </Link>
                )
            })}
        </div>
    )
}


export default SearchItems