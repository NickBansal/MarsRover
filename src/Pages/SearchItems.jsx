import React from "react";
import "../Stylesheets/SearchPage.css";
import { Link } from "@reach/router";
import MediaSelectForm from "../Components/MediaSelectForm";
import SingleThumbnail from "../Components/SingleThumbnail";

const SearchItems = ({ allItems, handleClick, input }) => {
  let allItemsCopy = allItems.filter(data => data.data[0].media_type === input);
  const filterMessage =
    input === "video"
      ? "There are no videos"
      : input === "audio"
      ? "There is no audio"
      : "There are no images";
  return (
    <div>
      {allItems.length > 0 && (
        <div className="Results">
          <MediaSelectForm handleClick={handleClick} />
          <p>Results: {allItemsCopy.length}</p>
        </div>
      )}

      {allItemsCopy.length === 0 && (
        <p className="EnterSearch">{filterMessage} available for this search</p>
      )}

      {allItemsCopy.map(items => {
        const { title } = items.data[0];
        const nasaId = items.data[0].nasa_id;
        const { href } = items.links[0];
        const newTitle =
          title.length < 37 ? title : `${title.substring(0, 37)}...`;
        return (
          <Link key={nasaId} to={`/search/${nasaId}`}>
            <SingleThumbnail href={href} title={title} newTitle={newTitle} />
          </Link>
        );
      })}
    </div>
  );
};

export default SearchItems;
