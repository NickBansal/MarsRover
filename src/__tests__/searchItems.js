import '../setupTest';
import React from 'react'
import SearchItems from '../Pages/SearchItems';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import fakeData from '../__mocks__/testData'


describe('<SearchPage />', () => {
    let wrapper
    
    beforeEach(() => {
        const SearchProps = {
            allItems: fakeData.collection.items,
            handleClick: () => {},
            input: ''
        }
        wrapper = shallow(<SearchItems {...SearchProps}/>)
    })
    it('Component matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    // it('Changes input props matching the slected value', () => {

        // expect(wrapper.find('input#image').length).toBe(2)
        // wrapper.find('input#video').simulate('click');
        // console.log(wrapper.props())
        // expect(wrapper.props()).toBe(0)
    // })
})