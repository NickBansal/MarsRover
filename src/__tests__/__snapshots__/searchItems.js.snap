// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> Component matches the snapshot 1`] = `
<div>
  <div
    className="Results"
  >
    <form
      onClick={[Function]}
    >
      <div
        className="RadioSelect"
      >
        <input
          defaultChecked={true}
          id="image"
          name="gender"
          type="radio"
          value="image"
        />
        <label
          value="image"
        >
          Image
        </label>
      </div>
      <div
        className="RadioSelect"
      >
        <input
          id="video"
          name="gender"
          type="radio"
          value="video"
        />
        <label
          value="video"
        >
          Video
        </label>
      </div>
      <div
        className="RadioSelect"
      >
        <input
          id="audio"
          name="gender"
          type="radio"
          value="audio"
        />
        <label
          value="audio"
        >
          Audio
        </label>
      </div>
    </form>
    <p>
      Results: 
      0
    </p>
  </div>
  <p
    className="EnterSearch"
  >
    There are no images
     available for this search
  </p>
</div>
`;
